scripts/check_compliance.py

```diff
@@ -221,6 +221,9 @@ class Codeowners(ComplianceTest):
     def parse_codeowners(self, git_root, codeowners):
         all_files = []
         with open(codeowners, "r") as codeo:
+            if self.commit_range != CURRENT_COMMIT_RANGE:
+                logging.warning("Checking %s range against current %s file" %
+                                (self.commit_range, codeowners))
```

Comment:
Could add "...make sure it includes all commits that add files" maybe.

Reply:
Was this meant to be a `==` by the way?

If the commit range has just the tip commit, then that might miss checking other commits in the same PR that add files. I was assuming the message was there to warn against stuff like that, but I might be misunderstanding it.

Maybe this code could be moved to `Codeowners.run()`.

Reply:
Most checks in this script allow testing any range, even past or future ranges that don't finish with ..HEAD. Let's say for instance that you have a very large series and you want to check just one early part of it.

However when looking at a past or future range, the Codeowners check stays stuck to the present CODEOWNERS file which will be incorrect if for instance you renamed a directory in the later part of your large series.

You're right: the current test is wrong but '==' is not the fix :-) It's wrong because it will print a **spurious** warning to any user checking HEAD~n.. where n>1

Reply:
Codeowners.run() does look like a better place.

Reply:
fix this

Reply:
Ah, yeah, should probably use the `CODEOWNERS` from the end of the range.

Even that isn't bulletproof, because it treats the entire range as a single commit w.r.t. `CODEOWNERS`. Could add a file in commit 1 and add it to `CODEOWNERS` in commit 5, and it wouldn't complain. It's a lot better though.

Reply:
Could parse the output of `git show <end of range>:CODEOWNERS` maybe.

Reply:
> Even that isn't bulletproof, because it treats the entire range as a single commit 

BTW I have a work in progress to change that for checkpatch which does support checking commits one by one and should let remove this sort of approximation "for free". I digress.